File: io.openems.backend.common/src/io/openems/backend/common/jsonrpc/response/GetAlertingConfigResponse.java
package io.openems.backend.common.jsonrpc.response;

import java.util.UUID;

import com.google.gson.JsonObject;

import io.openems.common.jsonrpc.base.JsonrpcResponseSuccess;
import io.openems.common.utils.JsonUtils;

/**
 * Represents a JSON-RPC Response for 'getAlertingConfig'.
 *
 * <pre>
 * {
 *   "jsonrpc": "2.0",
 *   "id": "UUID",
 *   "result": {
 *     timeToWait: int
 *   }
 * }
 * </pre>
 */
public class GetAlertingConfigResponse extends JsonrpcResponseSuccess {

	private final int timeToWait;

	public GetAlertingConfigResponse(int timeToWait) {
		this(UUID.randomUUID(), timeToWait);
	}

	public GetAlertingConfigResponse(UUID id, int timeToWait) {
		super(id);
		this.timeToWait = timeToWait;
	}

	@Override
	public JsonObject getResult() {
		return JsonUtils.buildJsonObject() //
				.addProperty("timeToWait", this.timeToWait) //
				.build();
	}

}

File: ui/src/app/edge/settings/alerting/getAlertingConfigResponse.ts
import { JsonrpcResponseSuccess } from "src/app/shared/jsonrpc/base";

export interface AlertingState {
    timeToWait: number;
}

/**
 * JSON-RPC Response to "getSystemUpdateState" Request.
 * 
 * <p>
 * 
 * <pre>
 * {
 *   "jsonrpc": "2.0",
 *   "id": "UUID",
 *   "result": {
 *      timeToWait": number
 *   }
 * } 
 * </pre>
 */
export class GetAlertingConfigResponse extends JsonrpcResponseSuccess {

    public constructor(
        public readonly id: string,
        public readonly result: AlertingState
    ) {
        super(id, result);
    }
}